backend/events/views.py

```
@@ -290,6 +297,49 @@ def _services_total(selected_services):
+        return False
+    if target_list.capacity == 0:
+        return True
+    return target_list.subscription_count < target_list.capacity
+
+
+def _get_main_waiting_lists(event, lock=False):
+    lists_qs = event.lists.all()
+    if lock:
+        # Evaluate to acquire row-level locks on all involved lists
+        all_lists = list(lists_qs.select_for_update().order_by('id'))
+        main_list = next((l for l in all_lists if l.is_main_list), None)
```

Ruff (0.15.7)

[error] Ambiguous variable name: `l`

(E741)